### PR TITLE
Sorted countries and added providerIds to labels

### PR DESCRIPTION
**Pull request by ignasbernotas**

### CODE REVIEW

**Comment by shroomist** [comment]
two similar PRs? #149 ?

**Comment by ignasbernotas** [comment]
Separate tasks, separate PRs, #149 derives from this.

File: src/app/countries/index.js [hunk 1]
```diff
@@ -36,4 +36,27 @@ function getCountryCodeFromProposal (proposal: ProposalDTO): ?string {
...
+    }
+
+    const idLength = 10
+
+    let idLabel = country.id
+    if (country.id.length > idLength) {
+      idLabel = country.id.substring(0, 10) + '...'
+    }
+
+    country.label = country.label + ' (' + idLabel + ')'
```
Comment (Waldz) [inline src/app/countries/index.js]: Better dont mutate variables after

File: src/app/countries/index.js [hunk 2]
```diff
@@ -36,4 +36,27 @@ function getCountryCodeFromProposal (proposal: ProposalDTO): ?string {
   return proposal.serviceDefinition.locationOriginate.country
 }
 
-export {getCountryName, getCountryNameFromProposal, getCountryCodeFromProposal}
+function getCountryListFromProposals (proposals: Array<ProposalDTO>) {
```
Comment (Waldz) [inline src/app/countries/index.js]: strange it's not country list it's formated proposals
+
it's not `country.id` it's `proposal.id`

**Comment by Waldz** [comment]
Looks good, just added my observations

**Comment by donce** [comment]
Sorting doesn't seem to work, i.e. Germany nodes are separated:
screen shot 2018-05-10 at 13 14 37

File: src/app/countries/index.js [hunk 3]
```diff
@@ -36,4 +36,27 @@ function getCountryCodeFromProposal (proposal: ProposalDTO): ?string {
...
+  let countries = proposals.map((proposal) => {
+    let country = {
+      label: getCountryNameFromProposal(proposal),
+      id: proposal.providerId,
+      code: getCountryCodeFromProposal(proposal)
```
[the remark this answers is not in the document]
Comment (ignasbernotas) [reply src/app/countries/index.js]: It's used in CountrySelect.vue

[the remark this answers is not in the document]
Comment (ignasbernotas) [reply src/app/countries/index.js]: Already done in #149

File: test/unit/app/countries/index.spec.js [hunk 4]
```diff
@@ -66,4 +71,37 @@ describe('Countries', () => {
...
+    it('returns sorted list', () => {
+      const list = getCountryListFromProposals(proposals)
+      expect('0x0987654321').to.be.eql(list[0].id)
+      expect('0x1234567890').to.be.eql(list[1].id)
+    })
+
+    it('truncates provider IDs', () => {
+      const list = getCountryListFromProposals(proposals)
+      expect('Lithuania (0x12345678...)').to.be.eql(list[1].label)
+      expect('Australia (0x09876543...)').to.be.eql(list[0].label)
```
[the remark this answers is not in the document]
Comment (ignasbernotas) [reply test/unit/app/countries/index.spec.js]: fixed

**Comment by ignasbernotas** [comment]
Closing this PR in favor of #149

File: src/main/tray/menu-builder.js [hunk 5]
```diff
@@ -23,9 +23,12 @@ function getMenuItems (
   )
 
   const connectSubmenu = new TrayMenu()
-  proposals.forEach((proposal: ProposalDTO) => {
-    connectSubmenu.add(getCountryNameFromProposal(proposal), () => {
-      communication.sendConnectionRequest({providerId: proposal.providerId})
+
+  let countries = getCountryListFromProposals(proposals)
```
[the remark this answers is not in the document]
Comment (ignasbernotas) [reply src/main/tray/menu-builder.js]: fixed

**Pull request closed without merging**